spec/reporter_spec.rb

```diff
@@ -0,0 +1,73 @@
+require_relative '../lib/forspell/reporter'
+require_relative '../lib/forspell/loaders/base'
+require_relative 'shared_examples'
+require 'yaml'
+require 'fakefs/spec_helpers'
+
+RSpec.describe Forspell::Reporter do
+  include FakeFS::SpecHelpers
+
+  let(:reporter) { described_class.new(logfile: logfile, verbose: verbose, format: format) }
+  let(:file) { 'file.rb' }
+  let(:word) { Forspell::Loaders::Word.new(file, 5, 'typo') }
+  let(:error) { [word, ['type']] }
+  let(:error_data) { [{file: file, line: 5, text: 'typo', suggestions: ['type']}] }
+
+  let(:format) { 'readable' }
+  let(:logfile) { 'log.out' }
+  let(:verbose) { false }
+
+  before do
+    FakeFS do
+      FileUtils.touch logfile
+    end
+    reporter.file(file)
+  end
+
+  after do
+    FakeFS.deactivate!
+  end
+
+  describe '#parsing_error' do
+    it 'should output parsing error' do
+      reporter.parsing_error('message')
+      expect(File.read(logfile)).to eq("Parsing error in #{file}: message\n")
```

Comment:
It could be expressed this way (probably a bit more expressive):
```ruby
it { expect { ... }.to change { File.read(logfile) }.to "Parsing error in #{file}: message\n" }
```